src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.cs

```diff
@@ -560,15 +557,15 @@ private void RestoreCheckpoint(in Checkpoint checkpoint)
             }
 
             // Remove any key encodings added after the current checkpoint.
-            // This is only needed when rolling back key reads in the Strict conformance level.
-            if (_keyEncodingRanges != null && _itemsWritten > checkpoint.ItemsWritten)
+            // This is only needed when rolling back key reads in the Strict conformance mode.
+            if (_keyEncodingRanges != null && _itemsRead > checkpoint.ItemsWritten)
             {
                 int checkpointOffset = checkpoint.Offset;
                 _keyEncodingRanges.RemoveWhere(key => key.Offset >= checkpointOffset);
             }
 
             _offset = checkpoint.Offset;
-            _itemsWritten = checkpoint.ItemsWritten;
+            _itemsRead = checkpoint.ItemsWritten;
```

Comment:
Assuming the checkpoint isn't shared with the writer, would you want to rename the checkpoint property, too?